app/forms/state_file/id_grocery_credit_form.rb

```diff
@@ -0,0 +1,86 @@
+module StateFile
+  class IdGroceryCreditForm < QuestionsForm
+    attr_accessor :dependents_attributes
+    delegate :dependents, to: :intake
+
+    set_attributes_for(
+      :intake,
+      :household_has_grocery_credit_ineligible_months,
+      :primary_has_grocery_credit_ineligible_months,
+      :spouse_has_grocery_credit_ineligible_months,
+      :primary_months_ineligible_for_grocery_credit,
+      :spouse_months_ineligible_for_grocery_credit
+    )
+
+    validates :household_has_grocery_credit_ineligible_months, inclusion: { in: %w[yes no], message: :blank }
+
+    validates :primary_months_ineligible_for_grocery_credit,
+              numericality: {
+                greater_than_or_equal_to: 0,
+                message: :blank
+              },
+              if: -> { primary_has_grocery_credit_ineligible_months == "yes" }
+    validates :spouse_months_ineligible_for_grocery_credit,
+              numericality: {
+                greater_than_or_equal_to: 0,
+                message: :blank
+              },
+              if: -> { spouse_has_grocery_credit_ineligible_months == "yes" }
+
+    def initialize(intake = nil, params = nil)
+      super
+      if params.present?
+        @intake.assign_attributes(dependents_attributes: dependents_attributes.to_h)
+      end
+    end
+
+    def save
+      @intake.update!(attributes_to_save)
+    end
+
+    def attributes_to_save
+      base_attrs = attributes_for(:intake)
+      updated_dependent_data = dependents_attributes
+
+      # set household member "has months" answers to no if household "has months" answer is no
+      if base_attrs[:household_has_grocery_credit_ineligible_months] == 'no'
```

Comment:
Enums let us track yes/no/unfilled instead of just true/false, which I think is primarily useful when generating output for optional pages, though we don't check the unfilled value of this answer anywhere IIRC. Someone who's been on the team longer than me can give a better answer here :) I don't feel very strongly about it either way, mostly just following convention.

Reply:
@DrewProebstel for more context 

also @jenny-heath @embarnard @mpidcock @tahsinaislam if there was more reasoning for this choice (unfilled is similar to `nil`, no?)

Reply:
FWIW I think that the decision to move away from using "unfilled" rather than nil as a default value in the app is well outside the scope of this PR, so discussion on that should probably happen elsewhere for posterity & visibility